File: java/estudos-java/curso-android-master/javaOOP/src/Veiculo.java
public class Veiculo {
    public Veiculo(String tipo, String cor, Double largura, Double altura) {
        System.out.println("Criando um veículo do tipo " + tipo);
        System.out.println("O veículo é da cor " + cor);
        System.out.println("A largura do veículo é " + largura);
        System.out.println("A altura do veículo é " + altura);
    }

    public void primeiraMarcha() {
        System.out.println("Passando a primeira marcha");
    }

    public void segundaMarcha() {
        System.out.println("Passando a segunda marcha");
    }
}

File: kotlin/estudos-kotlin/android-master/kotlin-oop/src/Veiculo.kt
open class Veiculo : IVeiculo {
    private var tipo: String = ""
    private var cor: String = ""
    private var altura: Double = 0.0
    private var largura: Double = 0.0

    constructor(tipo: String, cor: String, altura: Double, largura: Double) {
        this.tipo = tipo
        this.cor = cor
        this.altura = altura
        this.largura = largura
    }

    override fun detalhesVeiculo(cor: String, tipo: String, altura: Double, largura: Double) {
        println("Tipo: $tipo")
        println("Cor: $cor")
        println("Altura: $altura")
        println("Largura: $largura")
    }

    override fun marchas() {
        println("Quantidade de marchas: 5")
    }

    override fun habilitacao(): String {
        return "N/A"
    }
}